http-body-util/src/empty.rs

```diff
@@ -1,28 +1,27 @@
 use bytes::Buf;
 use http_body::{Body, Frame, SizeHint};
 use std::{
-    convert::Infallible,
     fmt,
     marker::PhantomData,
     pin::Pin,
     task::{Context, Poll},
 };
 
 /// A body that is always empty.
-pub struct Empty<D> {
-    _marker: PhantomData<fn() -> D>,
+pub struct Empty<D, E> {
```

Comment:
Probably worth discussing: what if it still had a default? `Empty<D, E = Infallible>`? It seems useful, but there's probably some cons to consider too.

Reply:
I think that makes sense!